apps/api/sharedLibs/crawler/src/lib.rs

```diff
@@ -175,6 +201,79 @@ pub unsafe extern "C" fn filter_links(data: *const libc::c_char) -> *mut libc::c
     CString::new(result).unwrap().into_raw()
 }
 
+fn _parse_sitemap_xml(xml_content: &str) -> Result<ParsedSitemap, Box<dyn std::error::Error>> {
```

Comment:
Add rust testing using `#[cfg::test]` for this function and run it once in your env to verify that it works